lighthouse/src/main.rs

```diff
@@ -50,6 +52,10 @@ fn bls_library_name() -> &'static str {
     }
 }
 
+fn bls_hardware_acceleration() -> bool {
+    !cfg!(feature = "portable")
```

Comment:
This just shows whether the portable feature flag is enabled, which is a compile-time property. We want to check whether the user's CPU supports the instructions _at runtime_.

To do that, we can do something similar to the SHA-extension detection in `ethereum_hashing`:

https://github.com/sigp/ethereum_hashing/blob/b6b5ff2d8db972ee6453a710e82aee3c638d5d82/src/lib.rs#L127-L140

The CPU feature we are interested in on x86 is ADX. On ARM I think there's a different extension we should check for, we would have to check the `blst` docs/code to determine what it is

Reply:
Oh got it. Will have a fix!

Reply:
@michaelsproul updated, please review. On AMD it's also ADX.

Reply:
I was referring to ARM, i.e. aarch64, not AMD (which uses the x86_64 ISA)

See https://github.com/sigp/lighthouse/pull/6039#discussion_r1664981977 for a suggested fix